src/main/java/grondag/canvas/pipeline/Image.java

```diff
@@ -61,40 +61,23 @@ protected void open() {
 				GFX.texParameter(config.target, params[i], params[++i]);
 			}
 
-			if (config.target == GFX.GL_TEXTURE_2D_ARRAY || config.target == GFX.GL_TEXTURE_3D) {
-				GFX.texImage3D(config.target, 0, config.internalFormat, width, height, config.depth, 0, config.pixelFormat, config.pixelDataType, (ByteBuffer) null);
-			} else if (config.target == GFX.GL_TEXTURE_CUBE_MAP) {
-				for (int face = 0; face < 6; ++face) {
-					GFX.texImage2D(GFX.GL_TEXTURE_CUBE_MAP_POSITIVE_X + face, 0, config.internalFormat, width, height, 0, config.pixelFormat, config.pixelDataType, (ByteBuffer) null);
+			GFX.texParameter(config.target, GFX.GL_TEXTURE_MAX_LEVEL, config.lod);
+			GFX.texParameter(config.target, GFX.GL_TEXTURE_MIN_LOD, 0);
+			GFX.texParameter(config.target, GFX.GL_TEXTURE_MAX_LOD, config.lod);
+			GFX.texParameter(config.target, GFX.GL_TEXTURE_LOD_BIAS, 0.0F);
```

Comment:
I might be cautious, but is it fine to not wrap this with `if (config.lod > 0)` block?

Reply:
done.